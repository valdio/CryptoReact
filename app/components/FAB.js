import GLOBALS from '../lib/globals'
import React, {Component} from 'react'
import {
    Image,
    StyleSheet,
    TouchableHighlight,
} from 'react-native'
import PropTypes from 'prop-types'
import refresh from'../assets/images/ic_refresh.png'

export default class FAB extends Component {
    render() {
        const {refreshTrigger} = this.props
        return (
            <TouchableHighlight
                underlayColor={GLOBALS.COLOR.RED}
                style={styles.container}
                onPress={refreshTrigger}
            >
                <Image
                    source={refresh}
                    style={styles.image}
                />
            </TouchableHighlight>
        )
    }
}

FAB.propTypes = {
    refreshTrigger: PropTypes.func.isRequired
}

const styles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        backgroundColor: GLOBALS.COLOR.HEADER_COLOR,
        margin: 16,
        padding: 6,
        elevation: 6,
        shadowColor: GLOBALS.COLOR.BLACK,
        shadowOffset: {width: 0, height: 4},
        shadowOpacity: 0.8,
        shadowRadius: 2,
        width: 50,
        height: 50,
        borderRadius: 50 / 2,
        zIndex: 5,
        position: 'absolute',
        right: 0,
        bottom: 0
    },
    image: {
        alignSelf: 'center',
        width: 24,
        height: 24,
    },
    button: {}

})
