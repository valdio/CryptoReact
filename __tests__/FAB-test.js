import React from 'react'
import FAB from '../app/components/FAB'
import renderer from 'react-test-renderer'

test('<FAB/> Snapshot Test', () => {
    const tree = renderer.create(
        <FAB
            refreshTrigger={()=>{}}
        />).toJSON()
    expect(tree).toMatchSnapshot()
})








