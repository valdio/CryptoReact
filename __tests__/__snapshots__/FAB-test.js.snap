// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAB/> Snapshot Test 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#258AE7",
      "borderRadius": 25,
      "bottom": 0,
      "elevation": 6,
      "height": 50,
      "justifyContent": "center",
      "margin": 16,
      "padding": 6,
      "position": "absolute",
      "right": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
      "width": 50,
      "zIndex": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
